### PR TITLE
[WIP] Tweaking the README

### PR DESCRIPTION
**Pull request by GillesVandewiele**
This PR is an attempt to tweak the README in order to make it a bit more fancy. I have tried to keep all the same information and extend it with a `Getting started` section.

You can see the rendered README here: https://github.com/GillesVandewiele/tslearn/tree/readme

### CODE REVIEW

**Comment by rtavenar** [comment]
This is definitely more appealing than the existing one, congrats !

I ll have a few comments but nothing huge, will write them down later.

File: README.md [hunk 1]
```diff
@@ -1,121 +1,115 @@
...
-scikit-learn
-joblib
-numba
-```
-
-If you plan to use the `shapelets` module, `keras` and `tensorflow` should also be installed.
-`h5py` is required for reading or writing models using the hdf5 file format.
+<!-- Our title -->
+<div align="center">
+  <h3>tslearn </h3>
```
Comment (GillesVandewiele) [inline README.md]: This would ideally be larger, but Github puts a line under `<h1>` and `<h2>`. Even more ideally, this would be replaced with some logo :)

File: README.md [hunk 2]
```diff
@@ -1,121 +1,115 @@
...
+<!-- Draw horizontal rule -->
+<hr>
+
+<!-- Table of content -->
+
+| Section | Description |
+|-|-|
+| [Installing](#installation) | Installing the dependencies and tslearn |
+| [Getting started](#getting-started) | A quick introduction on how to use tslearn |
+| [Documentation](#documentation) | A link to our API reference and a gallery of examples |
```
Comment (GillesVandewiele) [inline README.md]: Optionally, we can leave out the Documentation section from the readme and just directly link to RTD

**Comment by GillesVandewiele** [comment]
> This is definitely more appealing than the existing one, congrats !
> 
> I ll have a few comments but nothing huge, will write them down later.

Thanks! I left 2 comments as well of things that could be improved (but for the larger text I don't know how to do it).

Also, Travis seems to be getting an TimeError..

File: README.md (hunk 1 above)
Comment (rtavenar) [reply README.md]: It's already great like that. I agree that with a logo, it would be better, but we'll see that later :)

File: README.md (hunk 2 above)
Comment (rtavenar) [reply README.md]: I guess we can keep it like that so that the numbered sections fit well.

**Comment by rtavenar** [comment]
So, once again, this is a great move. I just have one suggestion: in the previous version, we had a (pretty imperfect, I confess) list of available features. I think this is something we could have here also, probably formatted in a more appealing way (and maybe not targeting exhaustivity).

Will add detailed comments here and there too.

File: README.md [hunk 3]
```diff
@@ -1,121 +1,115 @@
...
+* PyPi: `python -m pip install tslearn`
+* Conda: `conda install -c conda-forge tslearn`
+* Git: `python -m pip install https://github.com/rtavenar/tslearn/archive/master.zip`
+
+In order for the installation to be successful, the required dependencies must be installed. For a more detailed guide on how to install tslearn, please see the [Documentation](https://tslearn.readthedocs.io/en/latest/?badge=latest#installation)
+
+# Getting started
+
+## 1. Getting the data in the right format
+tslearn expects the data to be 3D. The three dimensions correspond to the number of time series, the number of measurements per time series and the number of dimensions respectively (`n_ts, max_sz, d`). In order to get the data in the right format, different solutions exist:
```
Comment (rtavenar) [inline README.md]: I would change to:

"tslearn expects a time series dataset to be formatted as a 3D `numpy` array".

File: README.md [hunk 4]
```diff
@@ -1,121 +1,115 @@
...
-When using PyPI, C++ build tools should be available to perform installation.
+After getting the data in the right format, a model can be trained. Depending on the use case, tslearn supports different tasks: classification, clustering and regression. For an extensive overview of possibilities, check out our [gallery of examples](https://tslearn.readthedocs.io/en/latest/auto_examples/index.html).
 
-Using `pip` should also work fine:
-```bash
-python -m pip install tslearn
+```python3
+>>> from tslearn.shapelets import ShapeletModel
+>>> clf = ShapeletModel({3: 1})
+>>> clf.fit(X_sax, y)
```
Comment (rtavenar) [inline README.md]: I would never try to run a `Shapelet` model on quantized time series representation such as SAX. Maybe fitting the model on `X_norm` would make more sense.

File: README.md [hunk 5]
```diff
@@ -1,121 +1,115 @@
...
+>>> my_third_time_series = [1, 2, 4, 2, 2]
+>>> X = to_time_series_dataset([my_first_time_series,
+                                my_second_time_series,
+                                my_third_time_series])
+>>> y = [0, 1, 1]
 ```
 
-## Using PyPI
+## 2. Data preprocessing and transformations
+Optionally, tslearn has several utilities to preprocess the data. In order to facilitate the convergence of different algorithms, you can [normalize time series](https://tslearn.readthedocs.io/en/latest/gen_modules/tslearn.preprocessing.html#module-tslearn.preprocessing). Alternatively, in order to speed up training times, one can [resample](https://tslearn.readthedocs.io/en/latest/gen_modules/preprocessing/tslearn.preprocessing.TimeSeriesResampler.html#tslearn.preprocessing.TimeSeriesResampler) the data or apply a [piece-wise transformation](https://tslearn.readthedocs.io/en/latest/gen_modules/tslearn.piecewise.html#module-tslearn.piecewise), such as SAX.
```
Comment (rtavenar) [inline README.md]: For some people, normalizing has a very precise sense (make the series unit norm). Maybe scaling is a more general term. (In this case, maybe the transformed dataset should be renamed `X_scaled`).

File: README.md [hunk 6]
```diff
@@ -1,121 +1,115 @@
...
-It seems on some platforms `Cython` dependency does not install properly.
-If you experiment such an issue, try installing it with the following command:
+The documentation, including a gallery of examples, is hosted at [readthedocs](http://tslearn.readthedocs.io/en/latest/index.html).
 
-```bash
-python -m pip install cython
-```
+# Contributing
+
+If you would like to contribute to `tslearn`, please have a look at [our contribution guidelines](CONTRIBUTING.md). A list of interesting TODO's can be found [here](https://github.com/rtavenar/tslearn/issues?utf8=✓&q=is%3Aissue%20is%3Aopen%20label%3A%22new%20feature%22%20). **If you want other ML methods for time series to be added to this TODO list, do not hesitate to open an issue!**
```
Comment (rtavenar) [inline README.md]: Here, maybe we could add a link to the "new issue" page: https://github.com/rtavenar/tslearn/issues/new/choose

File: README.md (hunk 4 above)
Comment (GillesVandewiele) [reply README.md]: That's true, that does not make that much sense indeed... I'll change that!

**Comment by GillesVandewiele** [comment]
I added a table with available features and handled the comments.

Could you double check on the table that I have left nothing out and that I classified most of the classes correctly?

File: README.md [hunk 7]
```diff
@@ -1,121 +1,123 @@
...
+>>> my_third_time_series = [1, 2, 4, 2, 2]
+>>> X = to_time_series_dataset([my_first_time_series,
+                                my_second_time_series,
+                                my_third_time_series])
+>>> y = [0, 1, 1]
+```
 
-# Dependencies
+## 2. Data preprocessing and transformations
+Optionally, tslearn has several utilities to preprocess the data. In order to facilitate the convergence of different algorithms, you can [normalize time series](https://tslearn.readthedocs.io/en/latest/gen_modules/tslearn.preprocessing.html#module-tslearn.preprocessing). Alternatively, in order to speed up training times, one can [resample](https://tslearn.readthedocs.io/en/latest/gen_modules/preprocessing/tslearn.preprocessing.TimeSeriesResampler.html#tslearn.preprocessing.TimeSeriesResampler) the data or apply a [piece-wise transformation](https://tslearn.readthedocs.io/en/latest/gen_modules/tslearn.piecewise.html#module-tslearn.piecewise).
```
Comment (rtavenar) [inline README.md]: It's still written "normalize" in the text here :)

File: README.md [hunk 8]
```diff
@@ -1,121 +1,123 @@
...
+</p>
+
+<!-- Draw horizontal rule -->
+<hr>
+
+<!-- Table of content -->
+
+| Section | Description |
+|-|-|
+| [Installing](#installation) | Installing the dependencies and tslearn |
```
Comment (rtavenar) [inline README.md]: I'd rather use the term "Installation" here (as in the section header btw)

**Comment by rtavenar** [comment]
> I added a table with available features and handled the comments.
> 
> Could you double check on the table that I have left nothing out and that I classified most of the classes correctly?

I'd put "barycenters" in the metrics column, also I'd fuse "Integration" and "Conversion" together in a single cell, something like: "Conversion (here and there)"

File: README.md [hunk 9]
```diff
@@ -1,121 +1,123 @@
...
+```python3
+>>> from tslearn.neighbors import KNeighborsTimeSeriesClassifier
+>>> knn = KNeighborsTimeSeriesClassifier(n_neighbors=1)
+>>> knn.fit(X_scaled, y)
+>>> print(knn.predict(X_scaled))
+[0 1 1]
 ```
 
-## Using PyPI
+## 4. Evaluation and analysis
```
Comment (rtavenar) [inline README.md]: I'd rather call that section "More analyses" or something in the like.

File: README.md [hunk 10]
```diff
@@ -1,121 +1,123 @@
...
-The easiest way to install `tslearn` is probably via `conda`:
-```bash
-conda install -c conda-forge tslearn
+```python3
+>>> from tslearn.neighbors import KNeighborsTimeSeriesClassifier
+>>> knn = KNeighborsTimeSeriesClassifier(n_neighbors=1)
+>>> knn.fit(X_scaled, y)
+>>> print(knn.predict(X_scaled))
+[0 1 1]
 ```
```
Comment (rtavenar) [inline README.md]: Here, I'd promote the fact that `tslearn` models are compatible with `sklearn` and hence cross validation and pipelines can be used (with a link to your gallery example).

**Comment by GillesVandewiele** [comment]
Handled the comments. No idea how the code coverage is decreasing... :joy:

**Comment by rtavenar** [comment]
LGTM

@johannfaouzi could you give an "external" opinion on this one?

File: README.md [hunk 11]
```diff
@@ -1,121 +1,124 @@
...
-This package builds on `scikit-learn`, `numpy` and `scipy` libraries.
-
-If you would like to contribute to `tslearn`, please have a look at [our contribution guidelines](CONTRIBUTING.md).
+# Installation
+There are different alternatives to install tslearn:
+* PyPi: `python -m pip install tslearn`
+* Conda: `conda install -c conda-forge tslearn`
+* Git: `python -m pip install https://github.com/rtavenar/tslearn/archive/master.zip`
+
+In order for the installation to be successful, the required dependencies must be installed. For a more detailed guide on how to install tslearn, please see the [Documentation](https://tslearn.readthedocs.io/en/latest/?badge=latest#installation)
```
Comment (johannfaouzi) [inline README.md]: Missing period at the end of the sentence.

File: README.md [hunk 12]
```diff
@@ -1,121 +1,124 @@
...
+
+# Getting started
+
+## 1. Getting the data in the right format
+tslearn expects a time series dataset to be formatted as a 3D `numpy` array. The three dimensions correspond to the number of time series, the number of measurements per time series and the number of dimensions respectively (`n_ts, max_sz, d`). In order to get the data in the right format, different solutions exist:
+* [You can use the utility functions such as `to_time_series_dataset`](https://tslearn.readthedocs.io/en/latest/gen_modules/tslearn.utils.html#module-tslearn.utils)
```
Comment (johannfaouzi) [inline README.md]: Missing period at the end of the sentence.

File: README.md [hunk 13]
```diff
@@ -1,121 +1,124 @@
...
+* [You can convert from other popular time series toolkits in Python](https://tslearn.readthedocs.io/en/latest/integration_other_software.html)
```
Comment (johannfaouzi) [inline README.md]: Missing period at the end of the sentence.

File: README.md [hunk 14]
```diff
@@ -1,121 +1,124 @@
...
+* [You can load any of the UCR datasets in the required format.](https://tslearn.readthedocs.io/en/latest/gen_modules/tslearn.datasets.html#module-tslearn.datasets)
+* [You can generate synthetic data using the `generators` module](https://tslearn.readthedocs.io/en/latest/gen_modules/tslearn.generators.html#module-tslearn.generators)
```
Comment (johannfaouzi) [inline README.md]: Missing period at the end of the sentence.

**Comment by GillesVandewiele** [comment]
Inserted the missing periods and made the headers smallers. The `###` do not get underlines on Github, so that is indeed better for the different steps in the `Getting started` section!
